### ISSUE TITLE
Investigate YOLayout for laying out CALayers

### ISSUE THREAD

**Issue by johnboiles** [YOLayout/YOLayout#40]
I think YOLayout would work just fine directly on a CALayer since CALayer has frame, layoutSubviews, setNeedsLayout etc. we should create YOLayer and try it out.

**Comment by johnboiles** [comment]
Tried this out on a view I was building that needed a custom animatable CALayer subclass. The view had normal subviews and also a custom CALayer subclass. I was able to call `[layout setFrame:view:]` on the layer just like I would a UIView and It Just Works™!

Interoperability between views and layers is a huge win. I've been hearing more and more people dropping down to CALayer level for their custom UI / animations. Keeping this issue open to create a YOLayer that mirrors YOView.

Sidenote: YOLayout could also be a good companion to [Facebook's AsyncDisplayKit](https://github.com/facebook/AsyncDisplayKit). Doesn't look like it'd be too hard to adapt since `ASDisplayNode` has all the relevant properties/methods except for `sizeThatFits:`. We'd have to add support for `ASDisplayNode`'s `- (CGSize)measure:(CGSize)constrainedSize;` method in addition to the normal `sizeThatFits:`
